src/NodeApi/Runtime/NativeLibrary.cs

```diff
@@ -101,10 +200,30 @@ private static nint dlopen(nint fileName, int flags)
     }
 
     [DllImport("libdl", EntryPoint = "dlopen")]
-    private static extern nint dlopen1(nint fileName, int flags);
+    private static extern nint dlopen1(string fileName, int flags);
 
     [DllImport("libdl.so.2", EntryPoint = "dlopen")]
-    private static extern nint dlopen2(nint fileName, int flags);
+    private static extern nint dlopen2(string fileName, int flags);
+
+    private static nint dlsym(nint handle, string name)
+    {
+        // Some Linux distros / versions have libdl version 2 only.
+        // Mac OS only has the unversioned library.
+        try
+        {
+            return dlsym2(handle, name);
+        }
+        catch (DllNotFoundException)
+        {
+            return dlsym1(handle, name);
+        }
+    }
+
+    [DllImport("libdl", EntryPoint = "dlsym")]
+    private static extern nint dlsym1(nint fileName, string flags);
```

Comment:
nit: the parameter names seems to be a bit incorrect: they should read `nint handle, string name`.

Reply:
Fixed.